.github/workflows/ci-cd.yml

```diff
@@ -6,46 +6,63 @@ name: CI-CD
 
 jobs:
   CI-CD:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.config.os }}
+
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      # we keep a matrix for convenience, but we would always want to run on one
+      # single OS and R version
```

Comment:
```suggestion
      # we keep a matrix for convenience, but we would typically just run on one
      # single OS and R version, aligned with the target deployment environment
```

Reply:
maybe it does not fit here as a comment, but it is not clear to me what a matrix implies and why is it desirable. Maybe worth adding a comment about this on the techguides